src/services/api.tsx

```diff
@@ -19,55 +100,182 @@ const getAge = (birthDate: Date) => {
   return age;
 };
 
-/* getCount function returns the number of resources of a type"; */
-export const getCount = async (resource: string, queryParameters: object) => {
-  const response = await client.search({
-    type: resource,
-    query: queryParameters
-  });
-  return response.data.total;
+export const getPatients = async (
+  param?: string,
+  count = PATIENT_REQUESTED
+) => {
+  /*
+   * getPatients return the lists of patients depending on param.
+   * param (optional): if empty, will return the whole list.
+   */
+
+  let response: PatientBundle = (await makeRequest(
+    "Patient",
+    false,
+    param,
+    count
+  )) as PatientBundle;
+
+  return addPatientsToBundle(response);
 };
 
-export const getPatients = async () => {
-  const response = await client.search({
-    type: "Patient",
-    query: { _count: 100 }
-  });
+export const getPatientsPerQuery = async (
+  searchName: String,
+  searchParams: any
+) => {
+  /*
+   * Function getPatientsPerQuery used to get PatientBundle according to query parameters
+   * searchNameParams: object containing name parameters search
+   * searchParams: object[] containing list of other parameters
+   * TODO : for now, this function is limited by the server response to 500 resources. Must find a way to improve it (searching computation done by the server ideally ?)
+   */
+  let bundles: Bundle[] = [];
+  let params: string;
```

Comment:
Could you initialize params to "" here instead of doing it several times below?

Reply:
you're totally right !